src/namepath.rs

```diff
@@ -14,6 +27,37 @@ impl<'src> Namepath<'src> {
       spaced: true,
     }
   }
+
+  pub fn push(&mut self, name: Name<'src>) {
+    self.0.push(name);
+  }
+
+  pub fn iter(&self) -> std::slice::Iter<'_, Name<'src>> {
+    self.0.iter()
+  }
+
+  pub fn len(&self) -> usize {
+    self.0.len()
+  }
+
+  /// Returns the last name in the module path
+  /// a::b::c -> c
+  /// a       -> a
+  pub fn last(&self) -> &Name<'src> {
+    self
+      .0
+      .last()
+      .expect("Internal error: Namepath can not be empty")
+  }
+
+  /// Splits a module path into the last name and its parent
+  /// a::b::c -> (a::b, c)
+  /// a       -> (empty, a)
+  pub fn split_last(&self) -> (&[Name<'src>], &Name<'src>) {
+    let name = self.last();
```

Comment:
Assigning this variable first instead of making this a one liner because `last()` will give a more meaningful panic message than `&self.0[..self.0.len() - 1]` panicking due to wrong indexing if `len()==0`